apitools/base/py/credentials_lib.py

```diff
@@ -48,65 +48,57 @@ def SetCredentialsCacheFileLock(lock):
     cache_file_lock = lock
 
 
-# TODO(craigcitro): Expose the extra args here somewhere higher up,
-# possibly as flags in the generated CLI.
+# List of additional methods we use when attempting to construct
+# credentials. Users can register their own methods here, which we try
+# before the defaults.
+_CREDENTIALS_METHODS = []
+
+
+def RegisterCredentialsMethod(method, position=None):
+    """Register a new method for fetching credentials.
+
+    This new method should be a function with signature:
+      client_info, **kwds -> Credentials or None
+    This method can be used as a decorator, unless position needs to
+    be supplied.
+
+    Note that method must *always* accept arbitrary keyword arguments.
+
+    Args:
+      method: New credential-fetching method.
+      position: (default: None) Where in the list of methods to
```

Comment:
so, two thoughts:

1. given that i don't forsee any other methods being added soon, i'm interested in keeping it simple.

2. in fact, i'm OK saying that how we add methods is an implementation detail, and waiting until i have a good reason to make it public.

so i'm making the methods private for now, and then i feel safer with the admittedly strange interface. if and when someone else presents a good case for sharing them, we'll see what we need.